src/applications/vaos/tests/mocks/form.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { combineReducers, applyMiddleware, createStore } from 'redux';
+import thunk from 'redux-thunk';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { fireEvent, waitFor } from '@testing-library/dom';
+
+import { commonReducer } from 'platform/startup/store';
+import { renderInReduxProvider } from 'platform/testing/unit/react-testing-library-helpers';
+
+import reducers from '../../reducers';
+import newAppointmentReducer from '../../reducers/newAppointment';
+
+import TypeOfCarePage from '../../containers/TypeOfCarePage';
+import { cleanup } from '@testing-library/react';
+
+export function createTestStore(initialState) {
+  return createStore(
+    combineReducers({
+      ...commonReducer,
+      ...reducers,
+      newAppointment: newAppointmentReducer,
+    }),
+    initialState,
+    applyMiddleware(thunk),
+  );
+}
+
+export async function setTypeOfCare(store, label) {
```

Comment:
I went back and forth about how much to mock out in these form page tests. All the pages depend on the page before them, so full integration tests would have to run through all the prior pages for each test of a given page. That seems like overkill. 

But I also don't want to mock a bunch of Redux state, because that greatly reduces my confidence in tests, because they're not necessarily reflective of how prior pages set up the state.

So this is my compromise. I want to create helper functions in here for the important pieces of data. These functions will take a Redux store, mount the appropriate page, set the data like a user would, by clicking on fields, then unmount the page. We're not running the full route-based app, but we're still mimicking what a user would do.

Reply:
I think this is a good compromise and a clever solution. Do you think by the time we get to the submit, we'd have a helper for each page?

Reply:
Probably not, but I'm not sure. Some of the pages have data that's independent enough from the rest of the form data that we wouldn't need to set it up. But I could also see us wanting to write a test that goes through the form and fully validates the data sent off at the end of the flow, and in that case, we probably would have helpers like this for each page. Then again, maybe it's easier to write that test in Cypress (in Nightwatch, you couldn't easily get the data POSTed to our mock api).

I think we'll end up with these helpers for type of care, the facility page, and the clinic page at least.